src/stronghold/secret.rs

```diff
@@ -129,6 +133,16 @@ impl SecretManage for StrongholdAdapter {
     }
 }
 
+#[async_trait]
+impl SecretManageExt for StrongholdAdapter {
+    async fn sign_transaction_essence(
+        &self,
+        prepared_transaction_data: &PreparedTransactionData,
+    ) -> crate::Result<Vec<UnlockBlock>> {
+        default_sign_transaction_essence(self, prepared_transaction_data).await
```

Comment:
Doesn't work

Reply:
Interesting, can you share the error ?

Reply:
```
no method named `default_sign_transaction_essence` found for reference `&mnemonic::MnemonicSecretManager` in the current scope
method not found in `&mnemonic::MnemonicSecretManager`
```
I removed it from 
```
impl<S> SecretManageExt for S
where
    S: SecretManage,
{
```
because it was conflicting with `sign_transaction_essence`, that's why it doesn't work

Changed it now

Reply:
Ah I see. Just impl `default_sign_transaction_essence` in `SecretManage` instead of having it take a dynamic dispatched parameter.

Reply:
Oh you did something similar while I was writing